check50/runner.py

```diff
@@ -159,60 +189,177 @@ def wrapper(run_root_dir, dependency_state):
             else:
                 result.passed = True
             finally:
+                # Register any newly created checks
+                if len(CHECK_REGISTRY.checks) > len(existing_checks) and not check._is_dynamic:
+                    raise _exceptions.Error(_("static check {} cannot create other checks, please mark it as dynamic with @check50.check(dynamic=True)".format(check.__name__)))
+                side_effects.new_checks = [c for c in CHECK_REGISTRY.checks if c not in existing_checks]
+
+                # In dynamic operation avoid serialization, store state in memory instead
+                if check._is_dynamic:
+                    CHECK_REGISTRY.dynamic_side_effects[check.__name__] = side_effects
+
+                    # Create new serializable side-effects to return to main process
+                    side_effects = CheckSideEffects(new_checks=side_effects.new_checks)
+
                 result.log = _log if len(_log) <= max_log_lines else ["..."] + _log[-max_log_lines:]
                 result.data = _data
-                return result, state
+                return result, side_effects
+
+        # Modules are evaluated from the top of the file down, so CheckRunner will
+        # contain checks in the order in which they are declared
+        CHECK_REGISTRY.register_check(wrapper)
+
         return wrapper
     return decorator
 
 
+class _CheckRegistry:
+    def __init__(self):
+        self.checks = []
+        self.static_side_effects = collections.defaultdict(CheckSideEffects)
+        self.dynamic_side_effects = collections.defaultdict(CheckSideEffects)
+        self._checks_in_memory = {}
+
+
+    def get_check(self, name):
+        return self._checks_in_memory[name]
+
+
+    def register_check(self, check):
+        self.checks.append({
```

Comment:
This should probably be an `attrs` class or at least a namedtuple rather than just a dict.